### PR TITLE
Check if connected to the internet before initiating DC service.

### PR DESCRIPTION
**Pull request by amoghbl1**

### CODE REVIEW

**Comment by phinfinity** [comment]
We want the app to be able to connect to DC hubs over 3G as well if required (public hubs on the internet), so we don't simply want to ensure wifi connectivity before allowing connections.
Although the problem here is that when a connection failed (wrong ip, unable to connect to hub) a meaning full message needs to be displayed.
The problem is in DCPPService.onHandleIntent() , the code there blindly catches exceptions (of connection failed) and continues as if it worked. This function needs to stop proceeding there and display some sort of Toast notification indicating failure.

**Comment by phinfinity** [comment]
closing this to solve the main problem at #21

**Comment by amoghbl1** [comment]
The second commit changes what I did in the first one, now I just check to see if it's connected to a network before starting the problem. Will continue at #21 !

File: src/com/phinmadvader/andcpp/LoginFragment.java [hunk 1]
```diff
@@ -41,11 +45,15 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
 		connectButton = (Button) rootView.findViewById(R.id.button);
 		connectButton.setText(ConnectString);
+		
+		
+		
 		connectButton.setOnClickListener(new OnClickListener() {
 			@Override
 			public void onClick(View view) {
-				if (!is_connected)
+				if (!is_connected){
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/LoginFragment.java:54]: again try not to make changes not relevant to a given change , in a single commit. (useful for book keeping and management later). (In this case although this change is useful , preferably in a different commit)

File: AnDCpp.iml [hunk 2]
```diff
@@ -4,6 +4,9 @@
     <facet type="android" name="Android">
       <configuration />
     </facet>
+    <facet type="android" name="Android2">
```
Comment (phinfinity) [inline AnDCpp.iml]: Don't make changes that are not relevant to the changes you're trying to make in a commit.
(Remove these changes, or make a separate commit if would be needed in a separate pull request. - In this case not needed to commit these configs)

File: src/com/phinmadvader/andcpp/LoginFragment.java [hunk 3]
```diff
@@ -132,4 +146,22 @@ public void connect() {
...
+	        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+	        return cm.getActiveNetworkInfo();
+	    }
+	    public static boolean isConnected(Context context){
+	    	Log.i("andcpp", "isConntectd?");
+	        NetworkInfo info = Connectivity.getNetworkInfo(context);
+	        return (info != null && info.isConnected());
+	    }
+	    public static boolean isConnectedWifi(Context context){
+	    	Log.i("andcpp", "isConnectedWiFi");
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/LoginFragment.java]: probably make this a debug log statement

File: src/com/phinmadvader/andcpp/LoginFragment.java [hunk 4]
```diff
@@ -132,4 +146,22 @@ public void connect() {
...
+	    public static NetworkInfo getNetworkInfo(Context context){
+	        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+	        return cm.getActiveNetworkInfo();
+	    }
+	    public static boolean isConnected(Context context){
+	    	Log.i("andcpp", "isConntectd?");
+	        NetworkInfo info = Connectivity.getNetworkInfo(context);
+	        return (info != null && info.isConnected());
+	    }
+	    public static boolean isConnectedWifi(Context context){
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/LoginFragment.java]: remove wifi specific function,  isConnected alone looks good!

**Comment by phinfinity** [comment]
Hey, I've looked at the part about checking network connectivity, and it does make sense to check this before attempting to connect. So it should be good to keep this! I've made some comments for some small changes. Also preferably meld both your commits into one. (You can ammend or rebase cherry pick in git to combine them)

File: AnDCpp.iml (hunk 2 above)
Comment (amoghbl1) [inline AnDCpp.iml]: sorry, that was an internal change made by android studio! I should have removed it...

**Comment by phinfinity** [comment]
just a small nit, (some stray unused code needs to be removed). But apart from that looks good! , Thanks for the patch!
